pkg/innerring/internal/blockchain/blockchain.go

```diff
@@ -0,0 +1,398 @@
+package blockchain
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/nspcc-dev/neo-go/pkg/config"
+	"github.com/nspcc-dev/neo-go/pkg/config/netmode"
+	"github.com/nspcc-dev/neo-go/pkg/consensus"
+	"github.com/nspcc-dev/neo-go/pkg/core"
+	"github.com/nspcc-dev/neo-go/pkg/core/storage"
+	"github.com/nspcc-dev/neo-go/pkg/core/storage/dbconfig"
+	"github.com/nspcc-dev/neo-go/pkg/core/transaction"
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	"github.com/nspcc-dev/neo-go/pkg/encoding/fixedn"
+	"github.com/nspcc-dev/neo-go/pkg/network"
+	"github.com/nspcc-dev/neo-go/pkg/rpcclient"
+	"github.com/nspcc-dev/neo-go/pkg/services/notary"
+	"github.com/nspcc-dev/neo-go/pkg/services/rpcsrv"
+	"github.com/nspcc-dev/neo-go/pkg/wallet"
+	utilConfig "github.com/nspcc-dev/neofs-node/pkg/util/config"
+	"go.uber.org/zap"
+)
+
+// Blockchain provides Neo blockchain services consumed by NeoFS Inner Ring
+// (hereinafter node). By design, Blockchain does not implement node specifics:
+// instead, it provides the generic functionality of the Neo blockchain, and
+// narrows the rich Neo functionality to the minimum necessary for the node's
+// operation.
+//
+// Blockchain must be initialized using New constructor. After initialization
+// Blockchain becomes a single-use component that can be started and then
+// stopped. All operations should be executed after Blockchain is started and
+// before it is stopped (any other behavior is undefined).
+type Blockchain struct {
+	logger    *zap.Logger
+	core      *core.Blockchain
+	netServer *network.Server
+	rpcServer *rpcsrv.Server
+
+	nodeAcc *wallet.Account
+
+	chErr chan error
+}
+
+// StorageConfig configures Blockchain storage.
+type StorageConfig struct {
+	typ  string
+	path string
+}
+
+// BoltDB configures Blockchain to use BoltDB located in given path.
+func BoltDB(path string) StorageConfig {
+	return StorageConfig{
+		typ:  dbconfig.BoltDB,
+		path: path,
+	}
+}
+
+// LevelDB configures Blockchain to use LevelDB located in given path.
+func LevelDB(path string) StorageConfig {
+	return StorageConfig{
+		typ:  dbconfig.LevelDB,
+		path: path,
+	}
+}
+
+// InMemory configures Blockchain to use volatile RAM storage.
+func InMemory() StorageConfig {
+	return StorageConfig{
+		typ: dbconfig.InMemoryDB,
+	}
+}
+
+// PingConfig configures P2P pinging mechanism.
+type PingConfig struct {
+	// Interval between pings. Must be positive.
+	Interval time.Duration
+
+	// Time period to wait for pong. Must be positive.
+	Timeout time.Duration
+}
+
+// P2PConfig configures communication over Neo P2P protocol.
+type P2PConfig struct {
+	// Specifies the minimum number of peers a node needs for normal operation. Must
+	// not be greater than math.MaxInt32.
+	MinPeers uint
```

Comment:
Can we have a dynamic default for it? Ceiling of (2/3 of committee size minus one).